### PR TITLE
CI: add julia-format action

### LINKED ISSUE #169: Add formatter

### PR DESCRIPTION
**Pull request by alyst**
This adds the automatic format check via [JuliaFormatter.jl](https://github.com/domluna/JuliaFormatter.jl) and GitHub code suggestions using [julia-format](https://github.com/julia-actions/julia-format) action.
This is an example of how it would look: https://github.com/alyst/StructuralEquationModels.jl/pull/2

Note that it would only suggest code formatting to the files modified by the PR, so to format the whole current codebase, a separate PR is required (it is as simple as running `using JuliaFormatter; format(".")` -- *JuliaFormatter* don't have to be a SEM.jl dependency, you can install it in the temporary Julia project; just the working folder should point to *SEM.jl*).
The formatting rules are specified in .JuliaFormatter.toml. Right now it is using the default formatting style.
Check the JuliaFormatter.jl docs for how to tune it to match to your preference.

I thought that it would be nice to fix the formatting before #181. One reason -- that PR contains many no-op whitespace changes (removal of trailing whitespace etc), which compilcate the review.

Fixes #169

### CODE REVIEW

**Comment by alyst** [comment]
This adds the automatic format check via [*JuliaFormatter.jl*](domluna/JuliaFormatter.jl) and GitHub code suggestions using julia-format action.
This is an example of how it would look: alyst/StructuralEquationModels.jl/pull/2

Note that it would only suggest code formatting to the files modified by the PR, so to format the whole current codebase, a separate PR is required (it is as simple as running `format(".")`).
The formatting rules are specified in `.JuliaFormatter.toml`. Right now it is using the default formatting style.
Check the *JuliaFormatter.jl* docs for how to tune it to match to your preference.

**Comment by Maximilian-Stefan-Ernst** [comment]
Nice, I now also added formatter to the test (so you are also told locally that you have to reformat before making a PR). Is there any reason why you are using `julia-format@master` instead of `julia-format@v3`?

**Comment by alyst** [comment]
> Is there any reason why you are using julia-format@master instead of julia-format@v3?

No, I was just exploring the logic of this action. AFAIK, v3 is not yet released, but v2 should do the same thing, except it would not fail if it detects format violations outside of the lines affected by the PR.
I would revert to v2.

File: .github/workflows/FormatCheck.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+name: Format suggestions
+on:
+  pull_request:
+    # this argument is not required if you don't use the `suggestion-label` input
+    types: [ opened, reopened, synchronize, labeled, unlabeled ]
+jobs:
+  code-style:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: julia-actions/julia-format@master
```
Comment (Maximilian-Stefan-Ernst) [inline .github/workflows/FormatCheck.yml:10]:
```suggestion
    steps:
      - uses: julia-actions/julia-format@v3
        with:
          version: '1'
          suggestion-label: 'format-suggest'
```

**Pull request merged**